hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -248,6 +248,19 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		}
 	}
 
+	// Set kubeadminPassword status
+	{
+		kubeadminPasswordSecret := manifests.KubeadminPasswordSecret(hcluster.Namespace, hcluster.Name)
+		err := r.Client.Get(ctx, client.ObjectKeyFromObject(kubeadminPasswordSecret), kubeadminPasswordSecret)
+		if err != nil {
+			if !apierrors.IsNotFound(err) {
```

Comment:
Nit: There is no need to nest this, can just chain with `err != nil && !apierrors.IsNotFound(err)`

Reply:
in that case the else would be executed if the error is not found, which I don't want. we could join the conditions first and then add a second check in the else part so that err == nil, but then that's just moving the extra check from one place to another.

Reply:
Oh you are right, I missed the `else` aspect of it